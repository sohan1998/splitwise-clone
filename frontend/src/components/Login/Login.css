.container {
    margin-top: 6rem;
    margin-left: 30rem;
}

.btn {
    margin-top: 0.7rem;
    background-color: #ff652f;
    color: white;
}

.inputLogin {
    font-size: 12pt;
    font-weight: 500;
}
