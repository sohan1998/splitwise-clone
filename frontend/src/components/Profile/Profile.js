import React, { Component } from 'react';
import '../Profile/Profile.css';
import axios from 'axios';
import Image from 'react-bootstrap/Image';
import { Button, Form, FormControl, ControlLabel } from 'react-bootstrap';
import backendServer from '../../webConfig';
import defaultpic from '../assets/default.jpg';

var useremail = localStorage.getItem('useremail');
var username = localStorage.getItem('username');

export default class Profile extends Component {
    constructor(props) {
        super(props);
        this.state = {
            useremail: useremail,
            username: username,
            // selectedFile: '',
            selectedFile: defaultpic,
            amazonurl: 'https://splitwisebucket1.s3.us-east-2.amazonaws.com/depositphotos_52374307-stock-illustration-blue-profile-icon.jpg',
        };
        // this.onFileChange=this.onFileChange.bind(this)
        this.onSubmit = this.onSubmit.bind(this);
        this.imageHandler = this.imageHandler.bind(this);
    }

    async componentDidMount() {
        var data = {
            useremail: this.state.useremail,
        };
        console.log('Email at frontend:', this.state.useremail);
        await axios.post(`${backendServer}/upload/getimageonload`, data).then((res) => {
            if (res.status === 200) {
                console.log('Got response', res.data.imagelink[0].profileImg);
                this.setState({ amazonurl: res.data.imagelink[0].profileImg });
            } else {
                console.log('There was some error!');
            }
        });
    }

    imageHandler = (e) => {
        console.log(e.target.files[0]);
        this.setState({ selectedFile: e.target.files[0] });
    };

    onSubmit = (e, file) => {
        const formData = new FormData();
        console.log('Inside submit data!');
        console.log('Got state of file:', file);
        formData.append('file', file);
        formData.append('useremail', useremail);
        axios.post(`${backendServer}/upload/imageupload`, formData).then((res) => {
            if (res.status === 200) {
                console.log('Image uploaded on S3!');
                var data = {
                    useremail,
                };
                console.log('Got email!', data.useremail);
                axios.post(`${backendServer}/upload/getimage`, data).then((res) => {
                    console.log('Inside frontend API!');
                    if (res.status === 200) {
                        console.log('Got response', res.data.imagelink[0].profileImg);
                        this.setState({ amazonurl: res.data.imagelink[0].profileImg });
                    } else {
                        console.log('There was some error!');
                    }
                });
            } else {
                console.log('There was some error!');
            }
        });
    };

    render() {
        return (
            <center>
                <div>
                    <h2>YOUR ACCOUNT</h2>
                </div>
                <br></br>
                <div>
                    <h5>Your name:</h5>
                </div>
                {this.state.username}
                <br></br>
                <br></br>
                <div>
                    <h5>Your email id:</h5>
                    {this.state.useremail}
                </div>
                <br></br>
                <div className='ImgUpload'>
                    <div className='myImage'>
                        <h5 className='heading'>Add your Image</h5>
                        <div className='form-group'>
                            <input type='file' id='file' accept='.png, .jpg, .jpeg' onChange={this.imageHandler} />
                        </div>
                        <div className='img-holder'>
                            {console.log(this.state.selectedFile)}
                            <img src={this.state.amazonurl} alt='' id='img' className='img' />
                        </div>
                        <div className='form-group'>
                            <button
                                className='btn btn-primary'
                                type='button'
                                onClick={(e) => {
                                    this.onSubmit(e, this.state.selectedFile);
                                }}
                            >
                                Upload
                            </button>
                        </div>
                    </div>
                </div>
            </center>
        );
    }
}
