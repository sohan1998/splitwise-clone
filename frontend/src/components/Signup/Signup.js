import React, { Component } from 'react';
import logo from '../assets/login_logo.png';
import '../Signup/Signup.css';
import axios from 'axios';
import { Redirect } from 'react-router';
import backendServer from '../../webConfig';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { signup } from '../../actions/actions.js';
var swal = require('sweetalert');

class Signup extends Component {
    constructor(props) {
        super(props);

        this.state = {
            name: null,
            email: null,
            password: null,
            error: '',
            userCreated: false,
        };

        this.handleuserCreate = this.handleUserCreate.bind(this);
        this.handleName = this.handleName.bind(this);
        this.handlePassword = this.handlePassword.bind(this);
        this.handleEmail = this.handleEmail.bind(this);
    }

    componentWillMount() {
        this.setState({ userCreated: false });
    }

    handleName = (e) => {
        this.setState({ name: e.target.value });
    };

    handleEmail = (e) => {
        this.setState({ email: e.target.value });
    };

    handlePassword = (e) => {
        this.setState({ password: e.target.value });
    };

    handleUserCreate = (e) => {
        var headers = new Headers();
        e.preventDefault();
        var data = {
            name: this.state.name,
            email: this.state.email,
            password: this.state.password,
        };
        this.props.signup(data);
    };

    componentWillReceiveProps(nextProps) {
        console.log('nextProps.ss.signupstatus', JSON.stringify(nextProps.ss.signupstatus));
        if (nextProps.ss.signupstatus) {
            console.log('nextProps.ss.signupstatus', nextProps.ss.signupstatus);
            // this.props.posts.unshift(nextProps.newPost);
        } else if (nextProps.ss.signupstatus == true) {
            swal('You are successfully signed up!', 'Please Login now.', 'success');
            this.setState({ userCreated: true });
        } else if (nextProps.ss.signupstatus == false) {
            this.setState({ userCreated: false });
        }
    }

    render() {
        var redirectVar = null;

        if (localStorage.getItem('signupcheck')) {
            console.log('Local storage found');
            redirectVar = <Redirect to='/login' />;
        }
        return (
            <div className='container'>
                {redirectVar}
                <div className='row'>
                    <img src={logo} class='col-md-3' height='240' alt='Splitwise'></img>
                    <div className='col-md-4'>
                        <h4
                            data-testid='SignupTest'
                            style={{
                                color: 'gray',
                                fontSize: 19,
                                marginBottom: 22,
                            }}
                        >
                            INTRODUCE YOURSELF
                        </h4>
                        <form>
                            <div className='form-group inputLogin'>
                                <label for='exampleInputName' style={{ fontSize: 30 }}>
                                    Hi there! My name is
                                </label>
                                <input
                                    type='text'
                                    class='form-control'
                                    id='exampleInputName'
                                    aria-describedby='emailHelp'
                                    placeholder='Enter Full Name'
                                    onChange={this.handleName}
                                />
                            </div>
                            <div className='form-group inputLogin'>
                                <label for='exampleInputEmail1'>
                                    Here's my&nbsp;
                                    <b>email address:</b>
                                </label>
                                <input
                                    type='email'
                                    className='form-control'
                                    id='exampleInputEmail1'
                                    aria-describedby='emailHelp'
                                    placeholder='Enter email'
                                    onChange={this.handleEmail}
                                />
                            </div>
                            <div className='form-group inputLogin'>
                                <label for='exampleInputPassword1'>
                                    And here's my&nbsp;
                                    <b>password:</b>
                                </label>
                                <input
                                    type='password'
                                    className='form-control'
                                    id='exampleInputPassword1'
                                    placeholder='Password'
                                    onChange={this.handlePassword}
                                />
                            </div>
                            <button type='submit' class='btn' onClick={this.handleUserCreate}>
                                Sign me up!
                            </button>
                            <p
                                style={{
                                    marginTop: 10,
                                    fontSize: 12,
                                }}
                            >
                                <a href='#'>By signing up, you accept the Splitwise Terms of Service.</a>
                            </p>
                        </form>
                    </div>
                </div>
                {redirectVar}{' '}
            </div>
        );
    }
}

Signup.propTypes = {
    signup: PropTypes.func.isRequired,
    ss: PropTypes.string,
};

const mapStateToProps = (state) => ({
    ss: state.usersignup,
});

export default connect(mapStateToProps, { signup })(Signup);
