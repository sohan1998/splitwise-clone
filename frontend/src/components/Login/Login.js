import React, { Component } from "react";
import logo from "../assets/login_logo.png";
import "../Login/Login.css";
import axios from 'axios';
import cookie from 'react-cookies';
import {Redirect} from 'react-router';
import backendServer from "../../webConfig";
import PropTypes from 'prop-types';
import { connect } from "react-redux";
import {login} from "../../actions/actions.js";
var loginsuccess = ""


class Login extends Component {
  constructor(props) {
    super(props);
    // this.handleLogout = this.handleLogout.bind(this);
    this.state = {
      email: null,
      password: null,
      authFlag: false,
      //emailerror:"",
      //passsworderror:"", 
  }
    this.emailChangeHandler = this.emailChangeHandler.bind(this);
    this.passwordChangeHandler = this.passwordChangeHandler.bind(this);
    this.submitOwnerLogin = this.submitOwnerLogin.bind(this);
    
  }

emailChangeHandler = (e) => {
  this.setState({
      email : e.target.value
  })
}
passwordChangeHandler = (e) => {
  this.setState({
      password : e.target.value
  })
}

submitOwnerLogin = (e) => {
  var headers = new Headers();
  console.log("Inside submitowner")
  e.preventDefault();
  const data = {
      email : this.state.email,
      password : this.state.password
  }
  this.props.login(data)
}


componentWillReceiveProps(nextProps) {
  console.log("JSON.stringify(nextProps)", JSON.stringify(nextProps))
  if (nextProps.ld.logindetails === true) {
    loginsuccess = "true"
    console.log("loginsuccess", loginsuccess)
    console.log("nextProps.logindetails", nextProps.ld.logindetails)
    //this.props.posts.unshift(nextProps.newPost);
  }
  else
    loginsuccess = "false"
}


  render() {
    var redirectVar = null;
    
    if(localStorage.getItem('myjwttoken') && localStorage.getItem('useremail') ){
          console.log("Local storage found")
          redirectVar = <Redirect to= "/dashboard"/>
        }
    return (
     <div className="container">
      {redirectVar}
        <div className="row">
          <img src={logo} class="col-md-3" height="240" alt="Splitwise"></img>
          <div className="col-md-4">
            <h4 data-testid="LoginTest" style={{ color: "gray", fontSize: 19, marginBottom: 22 }}>
              WELCOME TO SPLITWISE
            </h4>
            <form>
              <div class="form-group inputLogin">
                <label for="exampleInputEmail1">Email address</label>
                <input
                  type="email"
                  class="form-control"
                  id="exampleInputEmail1"
                  aria-describedby="emailHelp"
                  placeholder="Enter email"
                 onChange = {this.emailChangeHandler}
                 //errorText={this.state.emailerror}
                />
              </div>
              <div class="form-group inputLogin">
                <label for="exampleInputPassword1">Password</label>
                <input
                  type="password"
                  class="form-control"
                  id="exampleInputPassword1"
                  placeholder="Password"
                  onChange = {this.passwordChangeHandler}
                 // errorText={this.state.passsworderror}
                />
              </div>
              <button onClick={this.submitOwnerLogin} class="btn">
                Log in
              </button>
              <p style={{marginTop: 10}}>
                Forgot your password? <a href="#">Click here</a>
              </p>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

Login.propTypes = {
  login : PropTypes.func.isRequired,
  ld : PropTypes.string
}

const mapStateToProps = state => ({
ld : state.login
})

export default connect(mapStateToProps, {login})(Login); 