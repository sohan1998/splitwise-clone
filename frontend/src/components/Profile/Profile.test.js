/* eslint-disable react/prop-types */
/* eslint-disable no-undef */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
//import { createStore } from 'redux';
import { MemoryRouter } from 'react-router-dom';
// import configureStore from 'redux-mock-store';
//import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
//import allReducers from '../../reducers';
import NavbarClass from '../Navbar/Navbar';
import Login from '../Login/Login';
import Signup from '../Signup/Signup'
import Profile from '../Profile/Profile'
//const store = createStore(allReducers);

const component = TestRenderer.create(
  //<Provider store={store}>
    <MemoryRouter>
      <Profile>
    </Profile>
    </MemoryRouter>
 // </Provider>
);

// eslint-disable-next-line no-undef
afterEach(cleanup);

it('renders', async () => {
  expect(component.toJSON()).toMatchSnapshot();
});

test('Check for header', async () => {
  const { getByTestId } = render(
  //  <Provider store={store}>
      <MemoryRouter>
       <Profile>
        </Profile>
      </MemoryRouter>
    //</Provider>
  );
  expect(getByTestId('ProfileTest')).toHaveTextContent('Your name:');
});