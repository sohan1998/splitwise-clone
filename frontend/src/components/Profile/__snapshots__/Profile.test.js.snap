// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<center>
  <div>
    <h2>
      YOUR ACCOUNT
    </h2>
  </div>
  <br />
  <div>
    <h5
      data-testid="ProfileTest"
    >
      Your name:
    </h5>
  </div>
  
  <br />
  <br />
  <div>
    <h5>
      Your email id:
    </h5>
  </div>
  <br />
  <img
    className="account-img-height rounded"
    src="http://localhost:3001/untitled-3.jpg"
  />
  <input
    accept=".jpg"
    id="file"
    onChange={[Function]}
    type="file"
  />
  <button>
    upload
  </button>
</center>
`;